specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-06-01-preview/apimdiagnostics.json

```diff
@@ -0,0 +1,513 @@
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DiagnosticContract"
+          },
+          "description": "Page values."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Next page link if any."
+        }
+      },
+      "description": "Paged Diagnostic list representation."
+    },
+    "DiagnosticContract": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DiagnosticContractProperties",
+          "description": "Diagnostic entity contract properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./apimanagement.json#/definitions/Resource"
+        }
+      ],
+      "description": "Diagnostic details."
+    },
+    "SamplingSettings": {
+      "properties": {
+        "samplingType": {
+          "type": "string",
+          "description": "Sampling type.",
+          "enum": [
+            "fixed"
+          ],
+          "x-ms-enum": {
+            "name": "SamplingType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "fixed",
+                "description": "Fixed-rate sampling."
+              }
+            ]
+          }
+        },
+        "percentage": {
+					"type": "number",
+					"format": "double",
+					"description": "Rate of sampling for fixed-rate sampling."
+        }
+      }
+    },
+    "PipelineDiagnosticSettings": {
+      "properties": {
+        "request": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/HttpMessageDiagnostic"
+            }
+          ],
+          "description": "Diagnostic settings for request."
+        },
+        "response": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/HttpMessageDiagnostic"
+            }
+          ],
+          "description": "Diagnostic settings for response."
+        }
+      }
+    },
+    "HttpMessageDiagnostic": {
+      "properties": {
+        "headers": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Array of HTTP Headers to log."
+        },
+        "body": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/BodyDiagnosticSettings"
+            }
+          ],
+          "description": "Body logging settings."
+        }
+      }
+    },
+    "BodyDiagnosticSettings" : {
+      "properties": {
+        "bytes": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Number of request body bytes to log."
+        }
+      }
+    },
+    "DiagnosticContractProperties": {
+      "properties": {
+        "alwaysLog": {
+          "type": "string",
+          "description": "Specifies for what type of messages sampling settings should not apply. By now only the 'allErrors' value is supported."
```

Comment:
>description [](start = 11, length = 11)

we can define it an an enum, so it becomes easy in the client. Also set the modelAsString property to `true`, so it would not be considered breaking change, when you add more values.